### PR TITLE
feat: add vcctl jobtemplate

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
- Added vcctl jobtemplate command feature
  - create : Support `-f` (file) to pass in yaml file to create jobtemplate resource
  - delete: Support `-f` (file) to pass in yaml file to delete jobtemplate resource or `-N`(name) `-n`(namespace) to delete jobtemplate resource
  - describe: Support `-o, --format` to output in yaml or json format, `-N`(name) `-n`(namespace) to inquire 
  - list: Support  `-n`(namespace) to inquire 
  - get: Support  `-N`(name) ` `-n`(namespace) to inquire 
#### Which issue(s) this PR fixes:

Fixes https://github.com/volcano-sh/volcano/issues/3495

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?
Added vcctl jobtemplate command feature

```release-note
root@VM-0-9-ubuntu:~# vcctl jobtemplate delete -f jobtemplate.yaml
Deleted JobTemplate: default/a
Deleted JobTemplate: default/b
Deleted JobTemplate: default/c
Deleted JobTemplate: default/d
Deleted JobTemplate: default/e
Deleted 5 JobTemplates successfully

root@VM-0-9-ubuntu:~# vcctl jobtemplate create -f jobtemplate.yaml
Created JobTemplate: default/a
Created JobTemplate: default/b
Created JobTemplate: default/c
Created JobTemplate: default/d
Created JobTemplate: default/e

root@VM-0-9-ubuntu:~# vcctl jobtemplate get -N=a
Name    Namespace
a       default

root@VM-0-9-ubuntu:~# vcctl jobtemplate list
Name    Namespace
a       default
b       default
c       default
d       default
e       default

root@VM-0-9-ubuntu:~# vcctl jobtemplate describe -N=a
metadata:
  creationTimestamp: "2024-06-03T12:20:28Z"
  generation: 1
  managedFields:
  - apiVersion: flow.volcano.sh/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
...
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
None

```docs

```

### CODE REVIEW

File: cmd/cli/jobtemplate.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	}{
+		"create": {
+			Short: "create a jobtemplate",
+			RunFunction: func(cmd *cobra.Command, args []string) {
+				util.CheckError(cmd, jobtemplate.CreateJobTemplate(cmd.Context()))
+			},
+			InitFlags: jobtemplate.InitCreateFlags,
+		},
+		"list": {
+			Short: "list a jobtemplate",
```
Comment (Monokaix) [inline cmd/cli/jobtemplate.go:29]: -> list jobtemplates

File: pkg/cli/jobtemplate/jobtemplate_test.go [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-jobtemplate",
+							Namespace: "default",
+						},
+					},
+				},
+			},
+			Namespace:   "default",
+			ExpectedErr: nil,
```
Comment (Monokaix) [inline pkg/cli/jobtemplate/jobtemplate_test.go:40]: It's better to also test the print result, not just err: )

File: cmd/cli/jobtemplate.go (hunk 1 above)
Comment (googs1025) [reply cmd/cli/jobtemplate.go]: done

File: pkg/cli/jobtemplate/jobtemplate_test.go (hunk 2 above)
Comment (googs1025) [reply pkg/cli/jobtemplate/jobtemplate_test.go:42]: done

**Comment by william-wang** [comment]
@googs1025 It's great to have this feature. By the way, it's better to update the command line doc:)

**Review by william-wang — APPROVED** [review]
/approve

**Comment by googs1025** [comment]
> @googs1025 It's great to have this feature. By the way, it's better to update the command line doc:)

thanks @william-wang. I will add it to the issue to track

**Pull request merged**